flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -949,11 +982,73 @@ private void removeSlot(SlotID slotId) {
 					oldAllocationId,
 					new FlinkException("The assigned slot " + slot.getSlotId() + " was removed."));
 			}
+
+			removeSlotFromAllocatedResources(slotId);
 		} else {
 			LOG.debug("There was no slot registered with slot id {}.", slotId);
 		}
 	}
 
+	private void recordAllocatedSlotAndResource(TaskManagerSlot slot, ResourceProfile resourceProfile) {
```

Comment:
I think this method might only be called when
1. in `allocateSlot`.
2. in `registerSlot` and found that the allocationId is not null.